types.cc

```diff
@@ -1328,15 +1328,17 @@ list_type_impl::is_value_compatible_with_frozen(const collection_type_impl& prev
     return is_value_compatible_with_internal(*_elements, *lp._elements);
 }
 
-static void validate_aux(const list_type_impl& t, bytes_view v, cql_serialization_format sf) {
+template <FragmentedView View>
+static void validate_aux(const list_type_impl& t, View v, cql_serialization_format sf) {
     auto nr = read_collection_size(v, sf);
     for (int i = 0; i != nr; ++i) {
         t.get_elements_type()->validate(read_collection_value(v, sf), sf);
     }
-    if (!v.empty()) {
+    if (v.size_bytes()) {
+        auto hex = with_linearized(v, [] (bytes_view bv) { return to_hex(bv); });
```

Comment:
This might trigger a `std::bad_alloc` before we could throw `marshal_exception`. I think it is a bad idea to include the remains as it can be arbitrarily large. This is a pre-existing problem as we linearized before as well, on a higher level, so it can be done in a follow-up.

Reply:
Yes, I suppose we can include a few bytes before the problem area, bytes at the problem area, and a position indicator. We should be generous so that most inputs will be fully included, but no so generous that a 10MB map<> causes bad_allocs.